roamer.go

```diff
@@ -0,0 +1,147 @@
+// Package roamer flexible http request parser.
+package roamer
+
+import (
+	"context"
+	"net/http"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	"github.com/SLIpros/roamer/decoder"
+	roamerError "github.com/SLIpros/roamer/error"
+	"github.com/SLIpros/roamer/parser"
+	"github.com/SLIpros/roamer/value"
+)
+
+// TODO: Сохранять остаток json
+// TODO: Попробовать стандартный router
+// TODO: Поддержка файлов
+
+const (
+	SplitSymbol = ","
+)
+
+type Prepare interface {
```

Comment:
// AfterParser will be called after parsing.
type AfterParser interface {
	AfterParse(ctx context.Context) error
}